students/ericstreit/session03/list_lab.py

```diff
@@ -0,0 +1,79 @@
+#Lesson03
+#String Exercises ##
+#
+#!/usr/bin/env python3
+#define variables
+mylist = ["Apples", "Pears", "Oranges", "Peaches"]
+#define function
+def myfunc(n):
+    """Update the DocString"""
+    pass
+#series 1
+print("The list currently contains: ", mylist)
+additem = input("Pleaes enter another item: ")
+mylist.append(additem)
+print("The list currently now contains: ", mylist)
+l = len(mylist)
+listitem = int(input("Enter a list item from 1 to {} and I will show it to you: ".format(l)))
+print(mylist[(listitem-1)])
+print()
+pause = input("OK, let's add Kiwis to the list!")
+newlist = ["Kiwis"]
+mylist = newlist + mylist
+print(mylist)
+pause = input("Now adding Mangos to the list!")
+mylist.insert(0,"Mangos")
+print(mylist)
+pause = input("Now displaying all items that begin with the letter 'P'")
+for i in mylist:
+    if i[0] == "P":
+        print(i)
+pause = input("Press any key to continue to series 2 exercise")
+
+#series 2
+print()
+print()
+print("The list currently contains items: ",mylist)
+print("Removing the last item, {}, from the list ".format(mylist[-1]))
+mylist.pop()
+print("The list now contains: ",mylist)
+toremove = input("Please enter an item you would you like to remove : ")
+mylist.remove(toremove)
+print("The list now contains: ",mylist)
+
+#series 3
+#there is a bug in this section
+print("OK, let's move on to series 3!")
+print()
+for i in mylist:
+    choice = input("Do you like {}?: (y/n)".format(i))
+    if choice == "y":
+        continue
+    elif choice == "n":
+         mylist.remove(i)
+         print("Removing {} from the list!".format(i))
+         #bug if 'n' is selected it skips the next item in the list
```

Comment:
yup -- this is a common "gotcha" with mutable sequences (i.e. lists...):

if you mutate the list, particular if you remove something, the looping gets messed up -- "where am I in the list?"

so you want to loop through a copy, and mutate the original:

```
for i in mylist[:]:
   ...
```